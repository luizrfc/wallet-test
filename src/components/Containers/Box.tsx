import React from "react";
import { DimensionValue, View } from "react-native";
import { styleBox } from "./gridStyles";

interface IBox {
    width?: DimensionValue;
    pdHorizontal?: number;
    pdVertical?: number;
    children?: React.ReactNode;
}

const Box = ({
    width = "100%",
    pdHorizontal = 14,
    pdVertical = 10,
    children
}: IBox) => {
    return (
        <View style={styleBox(pdHorizontal, pdVertical, width).box}>
            {children}
        </View>
    );
};

export default Box;
