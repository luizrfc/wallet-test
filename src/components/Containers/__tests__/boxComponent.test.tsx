import React, { Fragment } from "react";
import "react-native";
import renderer from "react-test-renderer";
import Box from "../box";

test("Render box component correctly", () => {
  const button = renderer
    .create(
      <Box><Fragment></Fragment></Box>
    )
    .toJSON();
  expect(button).toMatchSnapshot();
});
