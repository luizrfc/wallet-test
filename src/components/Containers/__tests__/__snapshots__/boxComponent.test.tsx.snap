// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render box component correctly 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
/>
`;
